sql/hive/src/test/scala/org/apache/spark/sql/hive/InsertIntoHiveTableSuite.scala

```diff
@@ -494,4 +494,34 @@ class InsertIntoHiveTableSuite extends QueryTest with TestHiveSingleton with Bef
         spark.table("t").write.insertInto(tableName)
       }
   }
+
+  test(
+    """SPARK-20594: This is a walk-around fix to resolve a Hive bug. Hive requires that the
+      |staging directory needs to avoid being deleted when users set hive.exec.stagingdir
+      |under the table directory.""".stripMargin) {
+
+    withTable("test_table", "test_table1") {
+      spark.range(1).write.saveAsTable("test_table")
+
+      // Make sure the table has also been updated.
+      checkAnswer(
+        sql("SELECT * FROM test_table"),
+        Row(0)
+      )
+
+      sql("CREATE TABLE test_table1 (key int)")
+
+      // Set hive.exec.stagingdir under the table directory without start with ".".
+      sql("set hive.exec.stagingdir=./test")
+
+      // Now overwrite.
+      sql("INSERT OVERWRITE TABLE test_table1 SELECT * FROM test_table")
```

Comment:
We do not need `test_table`, right? 

`INSERT OVERWRITE TABLE test_table1 SELECT 1`